subprojects/core/src/integTest/groovy/org/gradle/api/resource/BrokenTextResourceIntegrationTest.groovy

```diff
@@ -38,6 +38,9 @@ class TextTask extends DefaultTask {
 
 task text(type: TextTask)
 """
+        executer.beforeExecute {
+            executer.requireOwnGradleUserHomeDir()
+        }
```

Comment:
Is the `beforeExecute` required? Can't we just call `super.requireOwnGradleUserHomeDir()`?

Reply:
Every time `executer` runs, the execution context will be cleared, so here it's used to keep the user home information because these tests contain more than one `run()` execution.

Reply:
Thanks, that's good to know!